### PR TITLE
PubOVL Spectator ESP feature

### PR DESCRIPTION
**Pull request by VierEck**
add Doctor Danks spectator esp controling code.
use Hitbox debug lines as esp overlay instead. no additional shader files needed. less resource heavy and better performance than Danks original spectator esp shader.
add setting for PubOVL spectator esp.

### CODE REVIEW

**Comment by VierEck** [comment]
it would look like this: 
https://www.youtube.com/watch?v=2Jd4pQ2wmdY

File: Sources/Client/Client.cpp [hunk 1]
```diff
@@ -63,10 +63,13 @@ DEFINE_SPADES_SETTING(cg_playerMessages, "1");
 DEFINE_SPADES_SETTING(cg_smallFont, "0");
 
 SPADES_SETTING(cg_playerName);
+DEFINE_SPADES_SETTING(dd_specWallhack);
 
 namespace spades {
 	namespace client {
 
+		Client* Client::globalInstance = nullptr;
```
Comment (Haxk20) [inline Sources/Client/Client.cpp:71]: global instance of client could very well be insecure. Its used in AreCheatsEnables (BTW HORRIBLE name. Like for real) and it is MUCH better to pass the client instance as pointer to the function. Global variables should be generally avoided as much as possible

File: Sources/Client/Client.h [hunk 2]
```diff
@@ -434,6 +434,8 @@ namespace spades {
 
 			void NetLog(const char* format, ...);
 
+			static Client* globalInstance;
```
Comment (Haxk20) [inline Sources/Client/Client.h:437]: Even if global instance of client was needed. You do not need to expose the global instance of the client via a header file.

File: Sources/Client/Client.h [hunk 3]
```diff
@@ -527,6 +529,9 @@ namespace spades {
 			void LocalPlayerHurt(HurtType type, bool sourceGiven, Vector3 source) override;
 			void LocalPlayerBuildError(BuildFailureReason reason) override;
 			// IWorldListener end
+
+			static bool AreCheatsEnabled(); // 'cheats', i.e. spectator wallhack or player names
```
Comment (Haxk20) [inline Sources/Client/Client.h:533]: As mentioned previously. Please change the names of the functions. They are not representative of what it is.

Its helpful for admins to check for players. So its not cheats.

WallhackActive -> SpectatorEspActive
AreCheatsEnabled -> AdminFeaturesEnabled

**Review by Haxk20 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**